### ISSUE TITLE
hh_client: An object of type vec does not allow array append

### ISSUE THREAD

**Issue by eryi** [facebook/hhvm#8331]
### HHVM Version
HipHop VM 3.28.2 (rel)
Compiler: 1538152486_476337476
Repo schema: c6deb0ae8a871a9b165c8053cd8f5dd7da48c2b7
hh--3.28.2 Sat Sep 29 00:37:30 2018

### Operating System and Version
Ubuntu 18.04

### Standalone code, or other way to reproduce the problem
```php
<?hh //strict

function test():void {
  $shape = shape(
    'vec' => vec[]
  );

  // no error
  $shape['vec'][] = "asd";

  if (false) {
    $shape['unrelated'] = '1234';
  }

  // an object of type vec does not allow array append (Typing[4006])
  $shape['vec'][] = "asd";
}
```

### Actual result
```
dev/bug.hh:16:3,17: an object of type vec does not allow array append (Typing[4006])
  dev/bug.hh:5:14,18: You might want to check this out
```

### Expected result
No errors!

**Comment by fredemmott** [comment]
Fixed in nightlies, will try to find the fix and cherry-pick

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones apply to the current versions of hhvm.

This issue has been fixed (tested in hhvm 4.56).
